modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -1348,6 +1356,75 @@ func statusError(err error) error {
 	return caddyhttp.Error(statusCode, err)
 }
 
+// delayClientGoneContext is a special context.Context type
+// intended for use when doing a RoundTrip where you don't
+// want a client disconnection to cancel the request during
+// the roundtrip. We delay the context.Done() channel until
+// the round trip is done, or until the parent context is
+// done, whichever comes first.
+type delayClientDoneContext struct {
+	context.Context
+	roundTripDone chan struct{}
+	done          chan struct{}
+	mu            sync.Mutex     // To protect access to the done channel
+	doneSet       atomic.Value   // To indicate whether done has been switched to ctx.Done
```

Comment:
Will `atomic.Bool` be a better choice?

Reply:
I believe that would be better, then i wouldn`t have to initialize it